bin/installHugo

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env sh
+
+# Hugo version is defined from the HUGO_VERSION environment variable
+# Default value is 0.88.1
+HUGO_VERSION="${HUGO_VERSION:-0.88.1}"
+
+case "$(uname)" in
+Linux) hugo_os='Linux-64bit' ;;
+Darwin) hugo_os='macOS-64bit' ;;
+*)
+  printf 'This script only works on Linux and Mac os\n' >&2
+  exit 1
+  ;;
+esac
+
+if ! command -v hugo >/dev/null 2>&1; then
+  # If hugo is not executable or not a file it must be downloaded and installed
+  hugo_install_path="$(cd "${0%/*}" && pwd)"
+  if ! { [ -x "${hugo_install_path}/hugo" ] && [ -f "${hugo_install_path}/hugo" ]; }; then
+    if ! {
+      tmp= && trap 'rm -fr -- "${tmp}"' EXIT INT &&
+        tmp="$(mktemp)" &&
+        hugo_archive="hugo_extended_${HUGO_VERSION}_${hugo_os}.tar.gz" &&
+        hugo_url="https://github.com/gohugoio/hugo/releases/download/v${HUGO_VERSION}/${hugo_archive}" &&
+        wget "${hugo_url}" -O "${tmp}" &&
+        cd "${hugo_install_path}" &&
+        tar -xzf "${tmp}" hugo
+    }; then
+      printf 'Failed to download hugo\n' >&2
+      exit 1
+    fi
+  fi
+
+  local_bin="${HOME}/.local/bin"
+  # Hugo needs be made available in local bin path
```

Comment:
Script works fine on my zsh config however I think making the bin available to the user's path should be an option : if I clone this repo and want the bin inside the repo to make some modifications and make sure everything works fine to do a PR then `rm -rf` the repo I don't want hugo still available on my system

Reply:
> Script works fine on my zsh config however I think making the bin available to the user's path should be an option : if I clone this repo and want the bin inside the repo to make some modifications and make sure everything works fine to do a PR then `rm -rf` the repo I don't want hugo still available on my system

If you `rm -rf` the repository, it removes hugo as well and it is no longer available to your system.
If the script find a hugo is already available in `$PATH` from your system or other installation; it does not make its own hugo available to the user and does not even create the symbolic link or `$PATH` entry.

There are 3 things that could possibly persist after deleting the repository with `rm -rf`, without any practical consequence at all:

1. The `~/.local/bin` directory (which is standard 'nix Linux or BSD path for user installed local shell commands/tools;
2. The `$PATH` environment variable entry for the `~/.local/bin` directory which is on the same standard practice as 1;
3. The now dead symbolic link `~/.local/bin/hugo` to the now deleted `path-to-repos/bin/hugo` executable. (which has no consequence at all, does not make hugo available, and only occupy a filesystem INode entry which is safe to keep there or delete later if one really need. Not necessary because root system paths are already and always full of dead symlinks from optionally installed tools or removed tools).

Anyway; What I could do if it can reinsure you is: split download and installation of hugo into two distinct scripts `downloadHugo` and `installHugo`. `installHugo` would call `downloadHugo` if needed. I feel it a bit overreaching needs for users of command-line shell tools but this is an option.

This initial script I updated is called `installHugo` so I improved and refined the install task, because it matches its explicit designation and logical intent.